azure-kusto-ingest/azure/kusto/ingest/kusto_ingest_client.py

```diff
@@ -138,56 +129,17 @@ def ingest_from_multiple_blobs(self, blobs, delete_sources_on_success, ingestion
         ingestion_properties : kusto_ingest_client.ingestion_properties.IngestionProperties
             The ingestion properties.
         """
-        self._refresh_queues_if_needed()
-        self._refresh_token_if_needed()
         for blob in blobs:
-            queue_details = random.choice(self._queues)
+            queues = self._resource_manager.get_ingestion_queues()
```

Comment:
Please be consistent. Either you choose randomly at KustoIngestClient, or you choose at ResourceManager.
IMHO, it's better to choose at KustoIngestClient, as we will probably later add a retry mechanism.